src/main/java/com/kelinci/mainapp/api/UserController.java

```diff
@@ -37,27 +31,38 @@ public String getListOfRegisteredUsers() {
 
     @DeleteMapping(value = "/registered/delete")
     public void deleteOurUser() {
-        registeredUser = null;
-        confirmedUser = null;
+        ourUser = null;
         listOfUsers.clear();
     }
 
     @PostMapping(value = "/registration")
     public void register(@RequestBody RegisterRequest request) {
 
-        Integer random = (int) (1000000 * Math.random());
-        String mailCode = Integer.toString(random);
-        System.out.println(mailCode);
-
-        final OurUser ourUser = new OurUser(request.getMail(), mailCode, false);
-
-        //wysylamy tutaj maila
-
-        registeredUser = ourUser;
-        listOfUsers.add(registeredUser);
-
+        final OurUser userFromRequest = new OurUser(request.getMail());
+        //in case the list is empty - add the userFromRequest with its randomly generated mailCode
+        if (listOfUsers.isEmpty()) {
+            String mailCode = generateMailCode();
+            userFromRequest.setMailCode(mailCode);
+            ourUser = userFromRequest;
+            listOfUsers.add(userFromRequest);
```

Comment:
A czy to nie dałoby się schować do prywatnej metody?

Reply:
wow ano można, tym bardziej że się tam dalej pojawia jeszcze raz - teraz kod jest śliczny.

Reply:
<3